[update] Troubleshooting section with the pg_dump error (#2240)
timescale/docs: _troubleshooting/self-hosted/pg_dump-permission-denied.md

---
title: Permission denied for table job_errors when running `pg_dump`
section: troubleshooting
products: [self_hosted]
topics: [backups]
errors:
  - language: bash
    message: |-
      pg_dump: error: query failed: ERROR: permission denied for table job_errors
      pg_dump: detail: Query was: LOCK TABLE _timescaledb_internal.job_errors IN ACCESS SHARE MODE
keywords: [backups, restore]
---

<!---
* Use this format for writing troubleshooting sections:
 - Cause: What causes the problem?
 - Consequence: What does the user see when they hit this problem?
 - Fix/Workaround: What can the user do to fix or work around the problem? Provide a "Resolving" Procedure if required.
 - Result: When the user applies the fix, what is the result when the same action is applied?
* Copy this comment at the top of every troubleshooting page
-->

>  The `pg_dump` tool tries to acquire lock on the job_errors table but
>  the user doesn't have the required SELECT permission resulting in the error.

Reviewer: ```suggestion

 When the `pg_dump` tool tries to acquire a lock on the `job_errors` 
 table, if the user doesn't have the required SELECT permission, it 
 results in this error.
```

>  A superuser needs to grant the necessary permission to the user who needs to
>  take the backup using the `pg_dump` tool.

Reviewer: ```suggestion
To resolve this issue, use a superuser account to grant the necessary 
permissions to the user requiring the `pg_dump` tool.
```

---

> title: Permission denied for table job_errors when running `pg_dump`

Reviewer 2: job_errors shows up as job _errors_ in the document. Can you add the backticks to job_errors as well?

Author: done!!!

section: troubleshooting
products: [self_hosted]
topics: [backups]
errors:
  - language: bash
    message: |-
      pg_dump: error: query failed: ERROR: permission denied for table job_errors
      pg_dump: detail: Query was: LOCK TABLE _timescaledb_internal.job_errors IN ACCESS SHARE MODE
keywords: [backups, restore]
---

<!---
* Use this format for writing troubleshooting sections:
 - Cause: What causes the problem?
 - Consequence: What does the user see when they hit this problem?
 - Fix/Workaround: What can the user do to fix or work around the problem? Provide a "Resolving" Procedure if required.
 - Result: When the user applies the fix, what is the result when the same action is applied?
* Copy this comment at the top of every troubleshooting page
-->

 When the `pg_dump` tool tries to acquire a lock on the `job_errors` 
 table, if the user doesn't have the required SELECT permission, it 
 results in this error.

To resolve this issue, use a superuser account to grant the necessary 

> permissions to the user requiring the `pg_dump` tool.

Reviewer 2: Is it possible to include the exact grant command required in the document?

```sql
GRANT SELECT ON TABLE _timescaledb_internal.job_errors TO <testuser>;
```

Author: Done!!!
